Update RealConnection.java

DevResRod opened the pull request:
Re-create the connection when Proxy server closes the connection after an unauthorized request, to avoid IO issue with closed socket.

swankjesse commented:
Please don’t combine behavior changes with refactorings. Ie. don’t extract methods.

Please include a test case that fails without the behavior change.

DevResRod commented:
I have reverted the method extraction (refactor) to ease the change tracking. Is it enought?
Also, added a new test method in CallTest.java that fails with the original code and succeeds with this commit.

yschimke commented on okhttp/src/main/java/com/squareup/okhttp/internal/io/RealConnection.java:
```diff
@@ -292,7 +293,16 @@ private void createTunnel(int readTimeout, int writeTimeout) throws IOException
-          if (tunnelRequest != null) continue;
+          if (tunnelRequest != null) {
+              String header = response.header("Connection");
+              if (header != null && header.toLowerCase().contains("close")) {
```
.contains() seems a bit weak here, value is generally keep-alive or close.

yschimke commented:
Any plans to migrate this to kotlin?  Seems like a useful fix, but not essential (it should recover anyway) unless you suffer this daily with additional latency and timeouts.
